### PR TITLE
Catch exception for PyPy

### PR DESCRIPTION
**Pull request by gabriel-munteanu**
## PR Summary

Catch additional exception, specific for PyPy (equivalent of Python 2.7). 
Because in python 2.7 `obj.__doc__` is read only, PyPy will throw an TypeError exception(somehow equivalent to AttributeError from CPython)

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant 
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/whats_new.rst if major new feature
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/cbook/deprecation.py [hunk 1]
```diff
@@ -174,8 +174,8 @@ def deprecate(obj, message=message, name=name, alternative=alternative,
             def finalize(wrapper, new_doc):
                 try:
                     obj.__doc__ = new_doc
-                except AttributeError:
-                    pass  # cls.__doc__ is not writeable on Py2.
+                except (AttributeError, TypeError):
+                    pass  # cls.__doc__ is not writeable on Py2. TypeError occurs on PyPy
```
Comment (tacaswell) [inline lib/matplotlib/cbook/deprecation.py]: This line is now too long :disappointed:

**Comment by tacaswell** [comment]
How close in Matplotlib to passing tests on pypy?

**Comment by gabriel-munteanu** [comment]
Fixed comment line being too long.
I will respond to you tomorrow about tests passing on pypy, when I get to my workstation

**Comment by anntzer** [comment]
You'll need to placate the style checker... see bottom of https://travis-ci.org/matplotlib/matplotlib/jobs/247129775

**Comment by gabriel-munteanu** [comment]
@tacaswell  Here is a pastebin with output from running matplotlib tests on PyPy: https://pastebin.com/uVeGPwuk

**Comment by tacaswell** [comment]
Skimming those issues a bunch of them look to be the pypy numpy not handling strings, a few places where we are checking refcounts, and one place where weakdicts work differently?

**Comment by gabriel-munteanu** [comment]
Yes, you are right. From my point of view the results of the tests are quite good.

**Comment by anntzer** [comment]
Can you force pypy to trigger the gc at some point (collecting *everything* that is unaccessible?)  That would solve the weakdict issue; the getrefcount issue could probably be solved in a similar manner.

**Comment by anntzer** [comment]
thanks

**Pull request merged**